nixos/modules/services/web-apps/opencloud.nix

```diff
@@ -0,0 +1,206 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib) types;
+  cfg = config.services.opencloud;
+  defaultUser = "opencloud";
+  defaultGroup = defaultUser;
+in
+{
+  options = {
+    services.opencloud = {
+      enable = lib.mkEnableOption "OpenCloud";
+
+      package = lib.mkPackageOption pkgs "opencloud" { };
+      webPackage = lib.mkPackageOption pkgs [ "opencloud" "web" ] { };
+      idpWebPackage = lib.mkPackageOption pkgs [ "opencloud" "idp-web" ] { };
+
+      configDir = lib.mkOption {
+        type = types.nullOr types.path;
+        default = "${config.services.opencloud.stateDir}/config";
+        defaultText = lib.literalExpression ''"''${config.services.opencloud.stateDir}/config"'';
+        example = "/var/lib/opencloud/config";
+        description = ''
+          Path to directory containing OpenCloud config file.
+
+          Example config can be generated by `opencloud init --config-path
```

Comment:
Can we run that automatically if the config file doesn't exist yet? Would be nice if simply setting `services.opencloud.enable = true` worked.

Reply:
Probably possible, yeah. I have not looked at the module stuff much yet, because there's a bunch of other upstream weirdness I'm chasing down.